### DISCUSSION TITLE
Frontend - Argument #3 ($newPassword) must be of type string, null given

### DISCUSSION THREAD

**Discussion by neolinux** [monarc-project/MonarcAppFO#523] [Q&A]
Hello,

i have this error when i try to change the default password:
![image](https://github.com/monarc-project/MonarcAppFO/assets/1484386/c4149615-aaba-4009-b340-3a7212fdb963)
`Fehler beim Abrufen von api/user/password/1: Monarc\Core\Service\PasswordService::changePassword(): Argument #3 ($newPassword) must be of type string, null given, called in /var/lib/monarc/releases/MonarcAppFO-v2.12.7-p2/vendor/monarc/frontoffice/src/Controller/ApiUserPasswordController.php on line 37 (500)`

What do i make wrong?

**Comment by ruslanbaidan** [comment]
The validation rules are not applied to the `Password change` functionality on the `My account` page.
A new password should comply the following rules:
`Your password is too short (min 9 characters).`
`Your password must contain at least: one uppercase, one number and one special character.`
The fields are both in red when it's e.g. too short, however there is no message and the button is active.
We are going to fix it as well as validate it on the Backend side and display a proper error message.
The error on the screen because the frontend filters the password out when it does comply to the rules, and sends empty data to the backend.

**Comment by ruslanbaidan** [comment]
The [fix](https://github.com/monarc-project/ng-client/pull/49) of the frontend part is going to be deployed within the next release or patch.
